### PR TITLE
fix bug with serializable range checking

### PR DESCRIPTION
**Pull request by carrino**

### CODE REVIEW

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java [hunk 1]
```diff
@@ -453,5 +454,42 @@ public void testCellReadWriteFailure2() {
...
+        put(t0, "row1", "col1", initialValue);
+        put(t0, "row1", "col2", initialValue);
+        put(t0, "row2", "col1", initialValue);
+        t0.commit();
+
+        Transaction t1 = startTransaction();
+        BatchingVisitables.copyToList(t1.getRange(TEST_TABLE, RangeRequest.builder().retainColumns(ImmutableList.of(PtBytes.toBytes("col1"))).build()));
+        get(t1, "row1", "col2");
+
+        // We need to do at least one put so we don't get caught by the
```
Comment (tmgordeeva) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java]: caught by the?

**Comment by joelea** [comment]
I've added appropriate release notes. We're going to put out a release with this in - trusting Tanya's +1 and the tests for the correctness portion. These classes are very difficult for anyone from London to code review for correctness in isolation right now.

**Comment by joelea** [comment]
Can someone also put a fix in for the comment when we know what it's supposed to be.

**Comment by carrino** [comment]
This issue came up when serializable transactions would throw without any contention.  The issue was that the range verification didn't take column selection into account.

**Pull request merged**